akka-docs/rst/scala/typed.rst

```diff
@@ -0,0 +1,321 @@
+supervisor we could either start the chat room from the gabbler (which we don’t
+want—it complicates its logic) or the gabbler from the chat room (which is
+nonsensical) or we start both of them from a third Actor—our only sensible
+choice:
+
+.. includecode:: code/docs/akka/typed/IntroSpec.scala#chatroom-main
+
+In good tradition we call the ``main`` Actor what it is, it directly
+corresponds to the ``main`` method in a traditional Java application. This
+Actor will perform its job on its own accord, we do not need to send messages
+from the outside, so we declare it to be of type ``Unit``. Actors receive not
+only external messages, they also are notified of certain system events,
+so-called Signals. In order to get access to those we choose to implement this
+particular one using the :class:`Full` behavior decorator. The name stems from
+the fact that within this we have full access to all aspects of the Actor. The
+provided function will be invoked for signals (wrapped in :class:`Sig`) or user
+messages (wrapped in :class:`Msg`) and the wrapper also contains a reference to
+the :class:`ActorContext`.
+
+This particular main Actor reacts to two signals: when it is started it will
+first receive the :class:`PreStart` signal, upon which the chat room and the
+gabbler are created and the session between them is initiated, and when the
+gabbler is finished we will receive the :class:`Terminated` event due to having
+called ``ctx.watch`` for it. This allows us to shut down the Actor system: when
+the main Actor terminates there is nothing more to do.
+
+Therefore after creating the Actor system with the ``main`` Actor’s
+:class:`Props` we just await its termination.
+
+Status of this Project and Relation to Akka Actors
+==================================================
+
+Akka Typed is the result of many years of research and previous attempts
+(including Typed Channels in the 2.2.x series) and it is on its way to
+stabilization, but maturing such a profound change to the core concept of Akka
+will take a long time. We expect that this module will stay experimental for
+multiple major releases of Akka and the plain ``akka.actor.Actor`` will not be
+deprecated or go away anytime soon.
+
+Being a research project also entails that the reference documentation is not
+as detailed as it will be for a final version, please refer to the API
+documentation for greater depth and finer detail.
+
+Main Differences
+----------------
+
+The most prominent difference is the removal of the ``sender()`` functionality.
+This turned out to be the Achilles heel of the Typed Channels project, it is
+the feature that makes its type signatures and macros too complex to be viable.
+The solution chosen in Akka Typed is to explicitly include the properly typed
+reply-to address in the message, which both burdens the user with this task but
+also places this aspect of protocol design where it belongs.
```

Comment:
Also removes the need for an Envelope, which is hard on the GC (at least long term, if we can remove the legacy Actor)